core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -152,28 +168,49 @@ class SnapshotService(protected val ctx: RawlsRequestContext,
         throw new RawlsExceptionWithErrorReport(ErrorReport(other))
     }
 
+  def enumerateSnapshotsByWorkspaceName(workspaceName: WorkspaceName,
+                                        offset: Int,
+                                        limit: Int,
+                                        referencedSnapshotId: Option[UUID] = None
+  ): Future[SnapshotListResponse] =
+    getV2WorkspaceContextAndPermissions(workspaceName,
+                                        SamWorkspaceActions.read,
+                                        Some(WorkspaceAttributeSpecs(all = false))
+    ).flatMap(workspaceContext =>
+      enumerateSnapshots(workspaceContext.workspaceIdAsUUID, offset, limit, referencedSnapshotId)
+    )
+
+  def enumerateSnapshotsById(workspaceId: String,
+                             offset: Int,
+                             limit: Int,
+                             referencedSnapshotId: Option[UUID] = None
+  ): Future[SnapshotListResponse] =
+    getV2WorkspaceContextAndPermissionsById(workspaceId,
+                                            SamWorkspaceActions.read,
+                                            Some(WorkspaceAttributeSpecs(all = false))
+    ).flatMap(workspaceContext =>
+      enumerateSnapshots(workspaceContext.workspaceIdAsUUID, offset, limit, referencedSnapshotId)
+    )
```

Comment:
in `enumerateSnapshotsByWorkspaceName` and `enumerateSnapshotsById`, there's extraneous `Future`-handling. See my comment below on having `enumerateSnapshots` NOT return a future. In that case, you can use `map` here instead of `flatMap`.

The difference between map and flatMap:
* `map`: chains `Future`s together, using the inner value of the last Future
* `flatMap`: chains `Future`s together and collapses nesting; e.g. `Future[Future[SnapshotListResponse]]` collapses into `Future[SnapshotListResponse]`

in `enumerateSnapshotsByWorkspaceName` and `enumerateSnapshotsById` you are calling two methods:
* `getV2WorkspaceContextAndPermissionsBy*` which returns `Future[Workspace]`
* `enumerateSnapshots ` which _currently_ returns `Future[SnapshotListResponse]`

with the current method signatures in this PR:
* `Future[Workspace] map Future[SnapshotListResponse]` chains the futures and returns `Future[Future[SnapshotListResponse]]` (which is the incorrect return type)
* `Future[Workspace] flatMap Future[SnapshotListResponse]` chains the futures and collapses the nesting, therefore returns `Future[SnapshotListResponse]` (which is correct)

_but_, if you change `enumerateSnapshots` to return `SnapshotListResponse` instead of `Future[SnapshotListResponse]`:
* `Future[Workspace] map SnapshotListResponse` chains the futures and returns `Future[SnapshotListResponse]` (which is correct)
* `Future[Workspace] flatMap SnapshotListResponse` does not compile, because flatMap expects another `Future` on its right-hand side

does this help?